packages/lesswrong/lib/collections/posts/helpers.ts

```diff
@@ -112,11 +110,9 @@ ${Posts.getLink(post, true, false)}
 };
 
 
-/**
- * @summary Get URL of a post page.
- * @param {Object} post
- */
+// @summary Get URL of a post page.
 Posts.getPageUrl = function(post: PostsMinimumForGetPageUrl, isAbsolute=false, sequenceId:string|null=null): string {
+  if (!post) return "/";
```

Comment:
This feels quite doomy to me, and also doesn't check out with the types. Like, if the types are right this whole check is impossible to occur. 

Why would we want this to be the default behavior of `getPageUrl`? This seems likely to cause some really tricky to spot bugs, and I prefer this function to throw an exception when you pass in an invalid post.

Reply:
I was about to write basically the same comment.

Reply:
Disagree. The situation was that, due to some upstream functions have inaccurate/incomplete annotations about nullability, you'd get into a situation where a URL was computed for a link, but it didn't actually get used. I expect this to come up in other situations as well, particularly in intermediate loading states.

Reply:
But the error state here is really horrible. I really don't want to accidentally have various things link to "/". There is absolutely no way I would catch that bug before I deploy it, and it would take an incredible amount of manual testing to catch this. 

At the very least, there should be a warning logged here. As it currently stands I assign more than 50% probability that within the next year, this will cause a bug that will cost me 2+ hours.

Reply:
Like, in the bug you are trying to address here, this would have broken the page in ways we would have never noticed until probably multiple weeks in.

Reply:
No, in the bug being addressed here, this code path triggers, but the url doesn't wind up getting used on the page.